LeagueOfLegendApi/lol/src/App.js

```diff
@@ -0,0 +1,103 @@
+import { useState } from "react";
+import "./App.css";
+import axios from "axios";
+import Header from "./Components/Header";
+import Tier from "./Components/Tier";
+
+function App() {
+  const [summonerName, setSummonerName] = useState("");
+  const [summonerNameData, setSummonerNameData] = useState({});
+  const [summonerSoloLeagueData, setSummonerSoloLeagueData] = useState([]);
+  const [summonerFlexLeagueData, setSummonerFlexLeagueData] = useState([]);
+  const ApiKey = "RGAPI-2e0de1b6-bda0-4c5d-84f5-19173b23ed63";
+  function searchForSummoner(event) {
+    let NameApiCallString =
+      "https://kr.api.riotgames.com/lol/summoner/v4/summoners/by-name/" +
+      summonerName +
+      "?api_key=" +
+      ApiKey;
+
+    //name 보냄 -> id, puuid 받음
+    axios.get(NameApiCallString).then(function (nameData) {
+      setSummonerNameData(nameData.data);
+      const summonerId = nameData.data.id;
+      const summonerPuuid = nameData.data.puuid;
```

Comment:
```suggestion
    const nameData = await axios.get(NameApiCallString);
    setSummonerNameData(nameData.data);
    const summonerId = nameData.data.id;
    const summonerPuuid = nameData.data.puuid;
```

async await 에 대해서 알아보는건 어떨까요?

async await 에 앞서 promise, callback 개념을 먼저 알아야 할 거 같아요.